sdk/helper/testhelpers/response_validation.go

```diff
@@ -0,0 +1,66 @@
+package testhelpers
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+// ValidateResponse validates whether the given response object conforms
+// to the response schema (schema.Fields). It cycles through the data map and
+// validates conversions in the schema. In "strict" mode, this function will
+// also ensure that the data map has all schema-required fields and does not
+// have any fields outside of the schema.
+//
+// This function is inefficient and is intended to be used in tests only.
+func ValidateResponse(schema *framework.Response, response *logical.Response, strict bool) error {
+	if response != nil {
+		return ValidateResponseData(schema, response.Data, strict)
+	}
+
+	return ValidateResponseData(schema, nil, strict)
+}
+
+// ValidateResponseData validates whether the given response data map conforms
+// to the response schema (schema.Fields). It cycles through the data map and
+// validates conversions in the schema. In "strict" mode, this function will
+// also ensure that the data map has all schema-required fields and does not
+// have any fields outside of the schema.
+//
+// This function is inefficient and is intended to be used in tests only.
+func ValidateResponseData(schema *framework.Response, data map[string]interface{}, strict bool) error {
```

Comment:
Same goes for this helper

Reply:
The test helpers could in theory be accepting testing.T, I was kind of on the fence between that and returning an error, but if folks think it would be a better style, I'm OK with switching it.

Reply:
I just like that to be explicit for shared testing package functions.

Reply:
Fair enough I can convert it.